### PR TITLE
Part 3 Transformer ranking. 

### PR DESCRIPTION
**Pull request by jobergum**
New day, new dog
image

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: _posts/2021-06-02-pretrained-transformer-language-models-for-search-part-3.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+Unlike the dense bi-encoder the contextualized late interaction model represents the query and document as multiple vectors obtained from the last output layer of the Transformer model. Bi-encoders on the other hand, usually performs a pooling operation over the last transformer layer, e.g just using the embedding representation from the *CLS* output token, or mean over all token output embeddings. Also, unlike other text to vector reresentations like [Word2Vec](https://en.wikipedia.org/wiki/Word2vec) the token vector representation depends on the other tokens in the same input sequence. For example the token *driver* in the text *Microsoft driver* has a different vector representation than *driver* in the text *Taxi driver* as the context is different. This thanks to the attention mechanism in the Transformer architecture where each token attends to all other tokens in the same input sequence.  We can say that token output vector representation is contextualized by the other tokens in the input text sequence.
```
Comment (kkraune) [inline _posts/2021-06-02-pretrained-transformer-language-models-for-search-part-3.md:26]:
```suggestion
Unlike the dense bi-encoder the contextualized late interaction model represents the query and document as multiple vectors obtained from the last output layer of the Transformer model. Bi-encoders on the other hand, usually performs a pooling operation over the last transformer layer, e.g just using the embedding representation from the *CLS* output token, or mean over all token output embeddings. Also, unlike other text to vector representations like [Word2Vec](https://en.wikipedia.org/wiki/Word2vec), the token vector representation depends on the other tokens in the same input sequence. For example the token *driver* in the text *Microsoft driver* has a different vector representation than *driver* in the text *Taxi driver* as the context is different. This thanks to the attention mechanism in the Transformer architecture where each token attends to all other tokens in the same input sequence.  We can say that token output vector representation is contextualized by the other tokens in the input text sequence.
```

File: _posts/2021-06-02-pretrained-transformer-language-models-for-search-part-3.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+> We denote the padding with masked tokens as query augmentation, a step that allows BERT to produce query-based embeddings at the positions corresponding to these masks. Query augmentation is intended to serve as a soft, differentiable mechanism for learning to expand queries with new terms or to re-weigh existing terms based on their importance for matching the query
```
Comment (kkraune) [inline _posts/2021-06-02-pretrained-transformer-language-models-for-search-part-3.md:30]:
```suggestion
> We denote the padding with masked tokens as query augmentation, a step that allows BERT to produce query-based embeddings at the positions corresponding to these masks. Query augmentation is intended to serve as a soft, differentiable mechanism for learning to expand queries with new terms or to re-weigh existing terms based on their importance for matching the query
```

File: _posts/2021-06-02-pretrained-transformer-language-models-for-search-part-3.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    num-threads-per-search: 1
+    first-phase {
+      expression: random 
+    }
+    summary-features {
+      onnxModel(colbert_encoder).contextual
+    }
+  }
+}
+
```
Comment (kkraune) [inline _posts/2021-06-02-pretrained-transformer-language-models-for-search-part-3.md:151]:
```suggestion
</pre>
```

**Review by kkraune — APPROVED** [review]
Great post!

**Comment by jobergum** [comment]
Thanks for the great review @kkraune

**Pull request merged**